crates/core/tedge_mapper/src/c8y_converter.rs

```diff
@@ -108,16 +116,53 @@ impl Converter for CumulocityConverter {
     }
 
     fn try_init_messages(&self) -> Result<Vec<Message>, ConversionError> {
-        let ops = Operations::try_new("/etc/tedge/operations")?;
-        let ops = ops.get_operations_list("c8y");
+        let fragments_message = create_inventory_fragments_message(&self.device_name)?;
+
+        let supported_operations_message = create_supported_operations_fragments()?;
 
-        let ops_msg = SmartRestSetSupportedOperations::new(&ops);
-        let topic = Topic::new_unchecked("c8y/s/us");
-        let msg = Message::new(&topic, ops_msg.to_smartrest()?);
-        Ok(vec![msg])
+        Ok(vec![supported_operations_message, fragments_message])
     }
 }
 
+fn create_supported_operations_fragments() -> Result<Message, ConversionError> {
+    let ops = Operations::try_new("/etc/tedge/operations")?;
+    let ops = ops.get_operations_list("c8y");
+
+    let ops_msg = SmartRestSetSupportedOperations::new(&ops);
+    let topic = Topic::new_unchecked("c8y/s/us");
+    Ok(Message::new(&topic, ops_msg.to_smartrest()?))
+}
+
+fn create_inventory_fragments_message(device_name: &str) -> Result<Message, ConversionError> {
+    let ops_msg = get_inventory_fragments(INVENTORY_FRAGMENTS_FILE_LOCATION)?;
+
+    let topic = Topic::new_unchecked(&format!(
+        "c8y/inventory/managedObjects/update/{}",
+        device_name
+    ));
+    Ok(Message::new(&topic, ops_msg.to_string()))
+}
+
+fn read_json_from_file(file_path: &str) -> Result<serde_json::Value, ConversionError> {
+    let mut file = File::open(Path::new(file_path))?;
+    let mut data = String::new();
+    file.read_to_string(&mut data)?;
+    let json: serde_json::Value = serde_json::from_str(&data)?;
+    Ok(json)
+}
+
+fn get_inventory_fragments(file_path: &str) -> Result<serde_json::Value, ConversionError> {
+    let mut json = read_json_from_file(file_path)?;
+
+    let agent_fragment = C8yAgentFragment::new();
+    let json_fragment = agent_fragment.to_json()?;
```

Comment:
This will error out of this function if the file read fails and won't create the right msg.

I believe this is place is to answer the question you asked in the PR, here you should handle the error case when the file is not read just create appropriate JSON struct from bare `c8y_Agent` fragment. There is no need to create the file, just document the location of the file.

eg:

```suggestion
    let agent_fragment = C8yAgentFragment::new();
    let json_fragment = agent_fragment.to_json()?;
    
    match read_json_from_file(file_path) {
     Ok(json) => ...
     Err(err) => ...
    };


```